### DISCUSSION TITLE
Built in component hints

### DISCUSSION THREAD

**Discussion by nandi95** [nuxt/nuxt#16476] [Ideas]
I'm not sure why I used to get typehints for nuxt 2 for example for `<NuxtLink />` but I'm not getting any for nuxt 3.
Maybe there was built in support from my IDE or a plugin....

Can we possibly add web-types to the project so IDEs will know what tags are available with what attributes?

Given there's only a couple components, this could literally be hand written.
I could do this if this is in fact a desired feature.

Resources:
https://blog.jetbrains.com/webstorm/2021/01/web-types/
https://json.schemastore.org/web-types

**Comment by danielroe** [comment]
Sounds like a good idea. I'm assuming we'd also need to do the same for auto-discovered components as well.

We currently declare GlobalComponents for this reason (consumed in Volar, for example):
https://github.com/nuxt/framework/blob/main/packages/nuxt3/src/components/templates.ts#L49-L60

**Reply by nandi95** [reply]
> auto-discovered components

I think web-types has to be noted in the `package.json` under the `web-types` key
not sure how the `.nuxt` folder can be targeted from a package in the `node_modules`

**Reply by nandi95** [reply]
Actually, I guess nuxt can just add the web-types key to its own package.json right?
Is there harm in that?
For reference, this is how I automatically generate them: https://github.com/karnama/vueish/blob/main/utils/web-types/buildWebTypes.ts
And just need to add the Lazy component options as an extra in there.

**Comment by pi0** [comment]
This feature was implemented in nuxt 2 components for web types suppot. We need to backport :)

**Reply by nandi95** [reply]
I haven't kept up on that part, has there been any changes to these components? Or is it literrally just copy this: https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/web-types/web-types.json ?

**Reply by nandi95** [reply]
@pi0 reminder of this question

**Reply by TechAkayy** [reply]
Would be great to have web-types generated for Nuxt 3. Thanks.
